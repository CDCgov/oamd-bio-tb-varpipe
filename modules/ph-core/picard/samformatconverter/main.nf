process PICARD_SAMFORMATCONVERTER {

    tag "${meta.id}"
    label 'process_medium'

    container 'oamd-bio-biocontainers-picard:3.3.0_47ba639_v2'

    input:
    tuple val(meta), path(alnSam)

    output:
    tuple val(meta), path("*.bam"), emit: bam
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def output_suffix = task.ext.output_suffix ?: 'bam'
    """
    picard SamFormatConverter \\
        INPUT=${alnSam} \\
        VALIDATION_STRINGENCY=${args} \\
        OUTPUT=${prefix}.${output_suffix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        picard: \$(echo \$(picard SamFormatConverter --version 2>&1) | grep -o 'Version:.*' | cut -f2- -d:)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_GATK.bam
    touch ${prefix}_GATK.bai

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        picard: \$(echo \$(picard SamFormatConverter --version 2>&1) | grep -o 'Version:.*' | cut -f2- -d:)
    END_VERSIONS
    """
}
