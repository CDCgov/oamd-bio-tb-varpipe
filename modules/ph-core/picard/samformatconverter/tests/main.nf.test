nextflow_process {

    name "Test Process PICARD_SAMFORMATCONVERTER"
    script "../main.nf"
    process "PICARD_SAMFORMATCONVERTER"
    config './nextflow.config'

    tag "modules"
    tag "modules_phcore"
    tag "picard"
    tag "picard/samformatconverter"

    test("tuberculosis aligned sam") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/bam/test.sam') ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("tuberculosis aligned sam -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file(params.modules_testdata_base_path + '/genomics/tuberculosis/bam/test.sam') ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
