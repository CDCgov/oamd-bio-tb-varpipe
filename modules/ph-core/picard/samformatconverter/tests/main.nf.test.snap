{
    "tuberculosis aligned sam -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_GATK.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3272f288e480acd1dc9d5bb75af39251"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test_GATK.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3272f288e480acd1dc9d5bb75af39251"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T11:15:22.231473"
    },
    "tuberculosis aligned sam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,0eee8853c194859c2c3b2770442da79c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3272f288e480acd1dc9d5bb75af39251"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,0eee8853c194859c2c3b2770442da79c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3272f288e480acd1dc9d5bb75af39251"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T11:26:51.118449"
    }
}
